contracts/connection-router/src/state.rs

```diff
@@ -101,6 +102,19 @@ pub struct Message {
     pub payload_hash: [u8; 32],
 }
 
+impl Message {
+    // TODO: pending to finalize the design of the message hash
+    pub fn hash_id(&self) -> [u8; 32] {
+        let mut hasher = Keccak256::new();
+        hasher.update(self.cc_id.to_string());
+        hasher.update(self.source_address.as_str());
```

Comment:
Need to be careful about things like casing and "0x" prefixing in addresses.

Reply:
lets discuss this in future PR